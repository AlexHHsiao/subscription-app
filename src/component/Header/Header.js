import React, {createRef, useCallback} from "react";
import AppConfig from "../../common/util/config";

const currencySelection = createRef();

const changeCurrency = () => {
    AppConfig.prevCurrency = AppConfig.currency;
    AppConfig.currency = currencySelection.current.value;
};

const Header = ({getPreview}) => {
    const onChange = useCallback(() => {
        changeCurrency();
        getPreview();
    }, [getPreview]);

    return (
        <nav className="navbar navbar-light" style={{backgroundColor: '#e3f2fd'}}>
            <span className="navbar-brand mb-0 h1">Subscription App</span>

            <form className="form-inline">
                <div className="input-group">
                    <div className="input-group-prepend">
                        <label className="input-group-text">Currency</label>
                    </div>
                    <select className="custom-select" onChange={onChange}
                            ref={currencySelection}>
                        <option value='USD'>USD</option>
                        <option value='CNY'>CNY</option>
                        <option value='HKD'>HKD</option>
                    </select>
                </div>
            </form>
        </nav>
    );
};

export default Header;
