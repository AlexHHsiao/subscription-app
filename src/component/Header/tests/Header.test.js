import React from 'react';
import {render} from '@testing-library/react';
import Header from '../Header';

test('<Header/>', () => {
    const {container: {firstChild}} = render(<Header getPreview={() => {}}/>);
    expect(firstChild).toMatchSnapshot();
});
