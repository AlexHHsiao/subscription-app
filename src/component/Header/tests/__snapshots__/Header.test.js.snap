// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> 1`] = `
<nav
  class="navbar navbar-light"
  style="background-color: rgb(227, 242, 253);"
>
  <span
    class="navbar-brand mb-0 h1"
  >
    Subscription App
  </span>
  <form
    class="form-inline"
  >
    <div
      class="input-group"
    >
      <div
        class="input-group-prepend"
      >
        <label
          class="input-group-text"
        >
          Currency
        </label>
      </div>
      <select
        class="custom-select"
      >
        <option
          value="USD"
        >
          USD
        </option>
        <option
          value="CNY"
        >
          CNY
        </option>
        <option
          value="HKD"
        >
          HKD
        </option>
      </select>
    </div>
  </form>
</nav>
`;
